src/python-common/ceph/deployment/service_spec.py

```diff
@@ -1227,16 +1234,6 @@ def _check_type(name: str, value: Optional[str], options: List[str]) -> None:
                 'Missing SNMP version (snmp_version)'
             )
 
-        _check_type('snmp_version',
-                    self.snmp_version,
-                    list(set(opt.value for opt in SNMPVersion)))
-        _check_type('auth_protocol',
```

Comment:
I can see the tests passing, but when I actually try and deploy bogus auth_protocol or privacy settings don't get caught. Maybe it's my local environment? Have you tested a deployment with bogus settings i.e. ceph orch apply -i <file> 
<pre>
service_type: snmp-gateway
service_name: snmp-gateway
placement:
  count: 1
spec:
  credentials:
    snmp_v3_auth_password: mypassword
    snmp_v3_auth_username: myuser
    snmp_v3_priv_password: mysecret
  port: 9464
  engine_id: 8000c53f0000000000
  privacy_protocol: WEIRD
  snmp_destination: 192.168.122.1:162
  auth_protocol: BIZARRE
  snmp_version: V3
</pre>